Library/Homebrew/extend/os/mac/missing_formula.rb

```diff
@@ -14,20 +19,33 @@ def blacklisted_reason(name)
         end
       end
 
-      def cask_reason(name, silent: false, show_info: false)
+      def cask_reason(name, silent: false, installed: false)
         return if silent
 
-        cask = Cask::CaskLoader.load(name)
-        reason = +"Found a cask named \"#{name}\" instead.\n"
-        if show_info
-          reason << Cask::Cmd::Info.get_info(cask)
-        else
-          reason << "Did you mean to type \"brew cask install #{name}\"?\n"
-        end
+        cask = find_cask(name, installed)
+        reason = +"Found a cask named \"#{name}\" instead. "
+        reason << suggest_command(cask, Homebrew.args.argv.first)
         reason.freeze
       rescue Cask::CaskUnavailableError
         nil
       end
+
+      def find_cask(name, installed)
+        return Cask::CaskLoader.load(name) unless installed
+
+        name if Cask::Caskroom.casks.any? { |cask| cask.to_s == name }
+        raise Cask::CaskUnavailableError, name
+      end
+
+      # how to handle multiple formulae where only one is installed as a cask?
+      def suggest_command(cask, command)
+        case command
+        when "install" || "uninstall"
+          "Try \"brew cask #{command} #{cask}\".\n"
```

Comment:
Use `<<~EOS` instead of hardcoding `n` and escaping `"`. I'd also recommend using backticks or a new line when suggesting commands.